lxd-agent/devlxd.go

```diff
@@ -117,7 +118,18 @@ var devlxdConfigKeyGet = devLxdHandler{"/1.0/config/{key}", func(d *Daemon, w ht
 }}
 
 var devlxdMetadataGet = devLxdHandler{"/1.0/meta-data", func(d *Daemon, w http.ResponseWriter, r *http.Request) *devLxdResponse {
-	client, err := getVsockClient(d)
+	var client lxd.InstanceServer
+	var err error
+
+	for i := 0; i < 10; i++ {
```

Comment:
It seems like we should do this also for the other uses of `getVsockClient` to handle the same issue.
How about changing `getVsockClient` to accept a `retries uint` argument and then define a constant that can be passed to it everywhere its used and push down this logic into `getVsockClient`?